model/construction_model.py

```diff
@@ -2,15 +2,12 @@
 from model.base_model import BaseModel
 from model.base_variables import BaseVariables
 from model.construction_objective import ConstructionObjective
-from utils import weights
 
 
 class ConstructionModel(BaseModel):
     def __init__(self, name, problem, data):
         super(ConstructionModel, self).__init__(name, problem, data, mip_focus=1, solution_limit=1)
 
-        self.weights = weights.get_weights()
```

Comment:
Is this intentional? From what I can see `self.weigths` is used on line 44.

Reply:
I would suggest removing the parameter from this class altogether, and rather just initialize this variable directly in `ConstructionObjective`

Reply:
Weights are initialized in base_model instead, so the intention is that both construction_model and optimality_model can use self.weights, as they both are built on the base_model. Correct e if this i wrong!

Reply:
Did not notice that, but that is a really good solution! Resolving this comment.